pkg/util/iptables/monitor_test.go

```diff
@@ -101,12 +102,43 @@ func (mfc *monitorFakeCmd) CombinedOutput() ([]byte, error) {
 		return []byte("iptables v1.6.2"), nil
 	}
 
-	if len(mfc.args) != 8 || mfc.args[0] != WaitString || mfc.args[1] != WaitSecondsValue || mfc.args[2] != WaitIntervalString || mfc.args[3] != WaitIntervalUsecondsValue || mfc.args[6] != "-t" {
-		panic(fmt.Sprintf("bad args %#v", mfc.args))
+	// Find the operation, chain name, and table name in the arguments
+	// The command structure can vary based on the iptables version and wait flags
+	// but will always have these components
+	var op operation
+	var chainName, tableName string
+	tableIndex := -1
+	opIndex := -1
+
+	// Find the table argument (-t)
+	for i, arg := range mfc.args {
+		if arg == "-t" && i+1 < len(mfc.args) {
+			tableName = mfc.args[i+1]
+			tableIndex = i
+			break
+		}
+	}
+
+	if tableIndex == -1 {
+		panic(fmt.Sprintf("no table specified in args %#v", mfc.args))
 	}
-	op := operation(mfc.args[4])
-	chainName := mfc.args[5]
-	tableName := mfc.args[7]
+
+	// Find the operation (like -N, -X, -F, etc.)
+	for i, arg := range mfc.args {
+		if i > 0 && strings.HasPrefix(arg, "-") && !strings.HasPrefix(arg, "--") &&
+			arg != "-t" && arg != "-w" && arg != "-W" && i != tableIndex+1 {
+			op = operation(arg)
+			opIndex = i
+			break
+		}
+	}
+
+	if opIndex == -1 || opIndex+1 >= len(mfc.args) {
+		panic(fmt.Sprintf("no operation found in args %#v", mfc.args))
+	}
+
+	// The chain name typically follows the operation
+	chainName = mfc.args[opIndex+1]
```

Comment:
we move from a statically hardcoded arguments match to a new dynamic logic to find the same parameters, @danwinship already did this for kube-proxy so he'll be the better to judge it, either works for me

/lgtm

Reply:
This isn't necessary. `monitorFakeExec`/`monitorFakeCmd` isn't intended to handle arbitrary iptables commands, it's intended only to handle the things that `Monitor` / `TestIPTablesMonitor` do. (eg, 99% of the `monitorFakeCmd` methods just panic, because the code "knows" that they won't be called.) All of the commands we have to parse are generated by `runContext()` from `iptables.go`, and it always puts the arguments in the same order.

Reply:
Ok, I didn't know the contexts. I see some UT fails in the monitorFakeExec/monitorFakeCmd when I removed the -W flag, this function CombinedOutput checks the `mfc.args[2] != WaitIntervalString || mfc.args[3] != WaitIntervalUsecondsValue`, so I made these changes. I can revert these changes, but I need to find a Linux machine to test it.